src/pygama/lgdo/array.py

```diff
@@ -1,46 +1,59 @@
+"""
+Implements a LEGEND Data Object representing an n-dimensional array and
+corresponding utilities.
+"""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
 import numpy as np
 
-from .lgdo_utils import *
+from pygama.lgdo.lgdo_utils import get_element_type
+
+log = logging.getLogger(__name__)
 
 
 class Array:
-    """Holds an ndarray and attributes
+    r"""Holds an :class:`numpy.ndarray` and attributes.
 
-    Array (and the other various array types) holds an "nda" instead of deriving
-    from ndarray for the following reasons:
+    :class:`Array` (and the other various array types) holds an `nda` instead
+    of deriving from :class:`numpy.ndarray` for the following reasons:
 
-    - it keeps management of the nda totally under the control of the user. The
-      user can point it to another object's buffer, grab the nda and toss the
-      Array, etc.
-    - it allows the management code to send just the nda's the central routines
-      for data manpulation. Keeping lgdo's out of that code allows for more
-      standard, reusable, and (we expect) performant python
-    - it allows the first axis of the nda to be treated as "special" for storage
-      in Tables
+    - It keeps management of the `nda` totally under the control of the user. The
+      user can point it to another object's buffer, grab the `nda` and toss the
+      :class:`Array`, etc.
+    - It allows the management code to send just the `nda`'s the central routines
+      for data manpulation. Keeping LGDO's out of that code allows for more
+      standard, reusable, and (we expect) performant Python.
+    - It allows the first axis of the `nda` to be treated as "special" for storage
+      in :class:`.Table`\ s.
 
     """
 
-
-    def __init__(self, nda=None, shape=(), dtype=None, fill_val=None, attrs={}):
+    def __init__(self, nda: np.ndarray = None, shape: tuple[int, ...] = (),
+                 dtype: np.dtype = None, fill_val: float | int = None,
+                 attrs: dict[str, Any] = {}) -> None:
```

Comment:
What do you think about this type for the `attrs` argument? I guess that would become very long if we starting specifying all the allowed types... or?

Reply:
I think `Any` is the right thing. It does look like in h5py the keys have to be strings (or maybe at least castable to strings)